core/tools/file/write.py

```
@@ -0,0 +1,32 @@
+from loguru import logger
+import os
+
+# 安全沙箱目录
+SANDBOX_DIR = Path(os.path.join(os.getcwd(), "agent_workspace")).resolve()
+SANDBOX_DIR.mkdir(parents=True, exist_ok=True)
+
+class WriteFileInput(BaseModel):
+    """写入文件输入"""
+    filepath: str = Field(..., description="文件路径（相对于 agent_workspace，例如：'main.py'）")
+    content: str = Field(..., description="要写入的内容")
+    mode: str = Field(default="w", description="写入模式：w=覆盖, a=追加")
```

Ruff (0.15.7)

[warning] String contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF001)